# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmOpIter.h
# Commit: 02ce4be46831
# Full Hash: 02ce4be46831d77239a749bd74b79a3e1447df0a
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 1
# Description:
#   Bug 1561521 - Refine ref.null decoding to decode heaptype instead of reftype. r=lth
#   
#   ref.null takes a heaptype immediate. This is backwards compatible for
#   reference-types, but is a breaking change for the GC proposal. This commit
#   implements decoding for this refinement.
# ==============================================================================

diff -r 00d2290648a7 -r 02ce4be46831 js/src/wasm/WasmOpIter.h
--- a/js/src/wasm/WasmOpIter.h	Tue Aug 18 00:40:20 2020 +0000
+++ b/js/src/wasm/WasmOpIter.h	Tue Aug 18 00:37:28 2020 +0000
@@ -487,7 +487,7 @@
   MOZ_MUST_USE bool readStructNarrow(ValType* inputType, ValType* outputType,
                                      Value* ptr);
   MOZ_MUST_USE bool readValType(ValType* type);
-  MOZ_MUST_USE bool readRefType(RefType* type);
+  MOZ_MUST_USE bool readHeapType(bool nullable, RefType* type);
   MOZ_MUST_USE bool readReferenceType(ValType* type, const char* const context);
 
 #ifdef ENABLE_WASM_SIMD
@@ -1658,7 +1658,7 @@
   MOZ_ASSERT(Classify(op_) == OpKind::RefNull);
 
   RefType type;
-  if (!readRefType(&type)) {
+  if (!readHeapType(true, &type)) {
     return false;
   }
   return push(type);
@@ -1681,8 +1681,8 @@
 }
 
 template <typename Policy>
-inline bool OpIter<Policy>::readRefType(RefType* type) {
-  return d_.readRefType(env_.types, env_.gcTypesEnabled(), type);
+inline bool OpIter<Policy>::readHeapType(bool nullable, RefType* type) {
+  return d_.readHeapType(env_.types, env_.gcTypesEnabled(), nullable, type);
 }
 
 template <typename Policy>